dspace-api/src/main/java/org/dspace/iiif/canvasdimension/IIIFCanvasDimensionServiceImpl.java

```diff
@@ -121,6 +121,7 @@ public void processItem(Context context, Item item) throws Exception {
                 }
             }
         }
+        context.uncacheEntity(item);
```

Comment:
To me, this seems like it might be in the wrong place... we only need to uncache if changes are made, correct?  Shouldn't this be inside the `if (isIIIFItem)` clause?

If you feel it really needs to be after every item, it might make more sense for this uncache call to move into the `while` loop inside `processCollection()`... as there's really no reason to uncache when you only process a *single* item (as this processItem() method can also be called standalone).  You just need to uncache when you process an entire Collection (via processCollection()).

Reply:
I wish I knew the answer to whether we only need to uncache if changes are made. (I haven't dug into how caching actually works.) Perhaps I can tell empirically by running on a large collection. 

From what I've seen, we are not successfully uncaching objects in the SAF item import process. For example, when I load collections of simple Items I need to limit the batch size to less than 1000 items and frankly less that 500 would be better. Otherwise the load becomes impossibly slow. And, if I load Items that contain many bitstreams, like a books with hundreds of bitstreams and OCR files, I need to limit the batch size to less than 100 items to have any hope of completing the migration. Given that Items with many bitstreams load dramatically slower that items with only a few, I suspect that uncaching Items as we do in the SAF processor is insufficient. 

That makes me want to be aggressive in uncaching everywhere...but obviously that's just guesswork on my part. In this case putting the uncache inside the `if (isIIIFItem)` clause does seem to make sense.

Reply:
BTW, I can create an issue for SAF item import if you agree there's a problem. 

On a related note (perhaps!) when I attempted to migrate an existing DSpace site to our new DSpace 7 instance using AIP's and packager I had to give up. I would have required over a week and a huge leap of faith to continue. I've been too busy to follow up and ask if others have seen this issue.

Reply:
@mspalti : If you are seeing a caching issue in SAF & have a way to reproduce it, then please do submit a ticket.  We keep doing our best to improve caching, and (as far as I was aware) batch processing should be working much smoother in 7.2 & 7.3.  But, if you are finding otherwise, then it'd definitely be good to get that into a bug report so that I can assign it to someone to analyze & try to reproduce & fix.

Reply:
OK. I believe I was using 7.2 on the production host but I'm not 100% sure about the version. I'll try to reproduce locally on the current main and submit an issue if necessary.

Reply:
Do we understand why Hibernate is not managing its caches effectively?  That's one of the things it is *for*.